src/Numeric/Sundials/CVode.hs

```diff
@@ -197,6 +199,19 @@ solveC CConsts{..} CVars{..} log_env =
   }
 
   while (1) {
+     // The solver will run until it terminates or receive a signal to stop by the
+     // way of a non null value in *ptrSTop
+     // The signal is an exception from outside (see the solve function in
+     // Numeric/Sundials.hs
+    // Ensure proper memory barrier so that `stopFlag` is not optimised away as constant
```

Comment:
what about `ptrStop` itself? Could you maybe add a comment explaining that `ptrStop` may also be considered as constant by the compiler and therefore optimized away?

Reply:
I've done edit of the comment to clarify that. Thank you.